File: 2021/day1/day1.py

def day1_01(lines: list):
    counter = 0
    for i in range(0, len(lines)):
        if (i == 0):
            print(lines[i] + " no previous measurment")
            continue
        if (lines[i] > lines[i-1]):
            counter += counter
            print(lines[i] + " increased")
        elif (lines[i] == lines[i-1]):
            print(lines[i] + " equal")
        else:
            print(lines[i] + " decreased")
    
        

forest = list()
# or forest = []
with open("./input.txt",'r') as f:
    for line in f:
        forest.append(line)

#print_forest(forest)
#print_forest(forest)
print ("puzzle 1 =", day1_01(forest))

File: rust_2021/day1/day1.js
fs = require("fs");

const lines = fs.readFileSync("input.txt", { encoding: "utf-8" }).split("\n");

let counter = 0;

for (let i = 0; i < lines.length-1; i++) {
    if (i == 0){
        console.log(lines[i] + " no previous measurment");
        continue;
    }
    if (lines[i] > lines[i-1]){
        counter++;
        console.log(lines[i] + " increased")
    }
    else if (lines[i] == lines[i-1]){
        console.log(lines[i] + " equal")
    }
    else 
        console.log(lines[i] + " decreased")
}

console.log(counter+1)

